### PR TITLE
let user know working directory will be used as content root path by default

### LINKED ISSUE #78789: Explictily log working directory and let user know it will be used as content root path by default.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

It's not the first time I making same mistake of starting published asp.net core application in docker without setting working directory:
```
ENTRYPOINT ["dotnet", "/app/api.dll"]
```
And content root path becomes `/`, so `appsettings` and other content files are not found. This causes app to fail on startup.

Host logs its content root path much later, so in case of failure I do not get a message like:
> Content root path: /app 

### Describe the solution you'd like

IMO, if host logged current working directory it would be much friendlier to the developer:

> Current working directory is `/`. If  Content root path is not set explicitly, then working directory is ued by default.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by pedrobsaila**
Fixes #78789

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.Hosting/src/Internal/ConsoleLifetime.cs [hunk 1]
```diff
@@ -79,6 +80,11 @@ private void OnApplicationStarted()
             Logger.LogInformation("Application started. Press Ctrl+C to shut down.");
             Logger.LogInformation("Hosting environment: {EnvName}", Environment.EnvironmentName);
             Logger.LogInformation("Content root path: {ContentRoot}", Environment.ContentRootPath);
+
+            if (Path.GetFullPath(Environment.ContentRootPath).Equals(Path.GetFullPath("."), StringComparison.InvariantCultureIgnoreCase))
```
Comment (kasperk81) [inline src/libraries/Microsoft.Extensions.Hosting/src/Internal/ConsoleLifetime.cs:84]: > Path.GetFullPath(".")

Envrionment.CurrentDirectory?

> InvariantCultureIgnoreCase

wrong on unix?

Comment (pedrobsaila) [reply src/libraries/Microsoft.Extensions.Hosting/src/Internal/ConsoleLifetime.cs]: > Envrionment.CurrentDirectory ?

Envrionment.CurrentDirectory when !DisableDefaults, and AppContext.BaseDirectory elsewhere

> InvariantCultureIgnoreCase
This project uses OrdinalIgnoreCase for string comparison so I'm using Ordinal to be unix compliant

**Review by jasper-d — COMMENTED** [review]
Could this be logged as info instead?

Some apps do not care about content root and would now need to explicitly configure it anyways or add extra logging config just to avoid that this warning pollutes observability systems.

File: src/libraries/Microsoft.Extensions.Hosting/src/Internal/ConsoleLifetime.cs [hunk 2]
```diff
@@ -79,6 +80,13 @@ private void OnApplicationStarted()
             Logger.LogInformation("Application started. Press Ctrl+C to shut down.");
             Logger.LogInformation("Hosting environment: {EnvName}", Environment.EnvironmentName);
             Logger.LogInformation("Content root path: {ContentRoot}", Environment.ContentRootPath);
+            string contentRootFullPath = Path.GetFullPath(Environment.ContentRootPath);
+
+            if (contentRootFullPath.Equals(System.Environment.CurrentDirectory, StringComparison.Ordinal)
+                || contentRootFullPath.Equals(AppContext.BaseDirectory, StringComparison.Ordinal))
```
Comment (steveharter) [inline src/libraries/Microsoft.Extensions.Hosting/src/Internal/ConsoleLifetime.cs:86]: When are these different?

Comment (pedrobsaila) [reply src/libraries/Microsoft.Extensions.Hosting/src/Internal/ConsoleLifetime.cs:86]: * when DisableDefaults = true, it is System.Environment.CurrentDirectory :

  * code :

https://github.com/dotnet/runtime/blob/d3f7d59e8dc44e5ce22c8ec9c011a1b72d7e9e92/src/libraries/Microsoft.Extensions.Hosting/src/HostingHostBuilderExtensions.cs#L216-L227

  * test :

https://github.com/dotnet/runtime/blob/d3f7d59e8dc44e5ce22c8ec9c011a1b72d7e9e92/src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostApplicationBuilderTests.cs#LL519C9-L524C10

* else it is AppContext.BaseDirectory :

  * code :

https://github.com/dotnet/runtime/blob/d3f7d59e8dc44e5ce22c8ec9c011a1b72d7e9e92/src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs#L230-L234

  * test :

https://github.com/dotnet/runtime/blob/d3f7d59e8dc44e5ce22c8ec9c011a1b72d7e9e92/src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostApplicationBuilderTests.cs#LL251C9-L283C10

Comment (pedrobsaila) [reply src/libraries/Microsoft.Extensions.Hosting/src/Internal/ConsoleLifetime.cs:86]: I'm not sure if there's a case when those two properties are different, but the two seems different by looking at the code

**Review by steveharter — APPROVED** [review]

**Comment by steveharter** [comment]
@dotnet-bot test

**Comment by steveharter** [comment]
There are a few CI errors; re-running

**Comment by steveharter** [comment]
@pedrobsaila there were concerns with this expressed in https://github.com/dotnet/runtime/issues/85809. Please see that issue. Thanks.

**Pull request merged**